zio-http/src/test/scala/zio/http/WebSocketSpec.scala

```diff
@@ -79,21 +79,25 @@ object WebSocketSpec extends HttpRunnableSpec {
 
         // Setup websocket server
 
-        serverHttp   = Http
-          .collectZIO[WebSocketChannelEvent] { case ChannelEvent(_, ChannelUnregistered) =>
-            isStarted.succeed(()) <&> isSet.succeed(()).delay(5 seconds).withClock(clock)
-          }
-          .toSocketApp
-          .toRoute
-          .deployWS
+        serverHttp   = Http.webSocket { channel =>
+          channel.receive.flatMap {
+            case Unregistered =>
+              isStarted.succeed(()) <&> isSet.succeed(()).delay(5 seconds).withClock(clock)
+            case _            =>
+              ZIO.unit
+          }.forever
+        }.toSocketApp.toRoute.deployWS
```

Comment:
`toSocketApp` can be deleted here and elsewhere, I think.

Reply:
This is the same issue as above. We need to go from an `Http` with input type of `WebSocketChannel` to an `Http` that no longer has this requirement so it can be further composed.